src/s3path.jl

```diff
@@ -8,13 +8,13 @@ end
 
 # constructor that converts but does not require type parameter
 function S3Path(segments, root::AbstractString, drive::AbstractString, isdirectory::Bool,
-                config::AbstractAWSConfig)
+                config::Union{Nothing, AbstractAWSConfig})
```

Comment:
I'd probably define a constant for this. Not certain what to call it but this seems fine:
```
const S3PathConfig = Union{AbstractAWSConfig, Nothing}
```